mitosheet/mitosheet/errors.py

```diff
@@ -405,6 +405,22 @@ def make_no_analysis_error(analysis_id: str, error_modal: bool=True) -> MitoErro
         error_modal=error_modal
     )
 
+def make_invalid_promote_row_to_header(error_modal: bool=True) -> MitoError:
+    """
+    Helper function for creating a invalid_promote_row_to_header.
+
+    Occurs when a user tries to promote a row to header that has row indexes
+    that are the same as this header.
+    """
+    to_fix = f'This row has values that overlap with row indexes. As such, making this row a header would make column references ambigious, so we cannot make this row the header row' 
```

Comment:
It also happens if the row has two values that are the same, which would also make column references ambiguous because the column headers would be the same. 

I actually think that might be the issue and it might have nothing to do with indexes. 

For example, take this dataframe `df = pd.DataFrame({'A': [1,2], 'B': [1,2]})` and try to promote the first row to the header. It will error.

Then update A1 -> 0, and try to promote it again. This time it will work. 

If you agree that is the issue, then lets update this error message because I think the explanation we can provide in that case could be way more clear and helpful.

Reply:
I added this example to https://github.com/mito-ds/monorepo/issues/267